substrate/primitives/consensus/beefy/src/lib.rs

```diff
@@ -318,16 +335,140 @@ impl<Number, Id, Signature> DoubleVotingProof<Number, Id, Signature> {
 	pub fn offender_id(&self) -> &Id {
 		&self.first.id
 	}
-	/// Returns the round number at which the equivocation occurred.
-	pub fn round_number(&self) -> &Number {
+}
+
+impl<Number, Id, Signature> BeefyEquivocationProof<Id, Number>
+	for DoubleVotingProof<Number, Id, Signature>
+{
+	fn offender_ids(&self) -> Vec<&Id> {
+		vec![self.offender_id()]
+	}
+
+	fn round_number(&self) -> &Number {
 		&self.first.commitment.block_number
 	}
-	/// Returns the set id at which the equivocation occurred.
-	pub fn set_id(&self) -> ValidatorSetId {
+
+	fn set_id(&self) -> ValidatorSetId {
 		self.first.commitment.validator_set_id
 	}
 }
 
+/// Proof of authority misbehavior on a given set id.
+/// This proof shows commitment signed on a different fork.
+/// See [`CheckForkEquivocationProof`] for proof validity conditions.
+#[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
+pub struct ForkEquivocationProof<Id: RuntimeAppPublic, Header: HeaderT, Hash> {
+	/// Commitment for a block on a different fork than one at the same height in
+	/// the chain where this proof is submitted.
+	pub commitment: Commitment<Header::Number>,
+	/// Signatures on this block
+	pub signatures: Vec<KnownSignature<Id, Id::Signature>>,
+	/// Canonical header at the same height as `commitment.block_number`.
+	pub canonical_header: Option<Header>,
+	/// Ancestry proof showing that the current best mmr root descends from another mmr root at
+	/// `commitment.block_number` than commitment.payload
+	pub ancestry_proof: Option<AncestryProof<Hash>>,
```

Comment:
From what I understand this works only if the equivocation is processed at the same block at which the ancestry proof was generated. If for some reason, the equivocation reporting extrinsic can't be added to the current block and it goes in a later one, this ancestry proof can't be used.

I think we should treat this case as well.

Reply:
Thanks, sorry, missed that